### PR TITLE
feat: migrate 1inch to v4

### PR DESCRIPTION
**Pull request by tima-t**
LIQ-685

### CODE REVIEW

**Review by bradleySuira — APPROVED** [review]
LGTM

File: src/store/getters.js [hunk 1]
```diff
@@ -43,6 +43,10 @@ const TESTNET_ASSETS = [
   })
 }, {})
 
+const mapLegacyProvidersToSupported = {
+  oneinchV3: 'oneinchV4'
```
Comment (kraikov) [inline src/store/getters.js:47]: Do we need `oneinchV3` at all now? Is `oneinchV4` supposed to completely replace `oneinchV3` ?

File: src/store/getters.js [hunk 2]
```diff
@@ -89,12 +93,13 @@ export default {
   },
   swapProvider (state) {
     return (network, providerId) => {
-      const cacheKey = [network, providerId]
+      const supportedProviderId = mapLegacyProvidersToSupported[providerId] ? mapLegacyProvidersToSupported[providerId] : providerId
```
Comment (kraikov) [inline src/store/getters.js:96]: Why is this needed now?

File: src/store/getters.js (hunk 1 above)
Comment (tima-t) [reply src/store/getters.js:47]: Yes, they are doing the same and V3 will be deprecated soon . 

I need this line, because we have swaps in the history that have been using V3 and they are in the local storage. 
If we do not have a backward support this will result in visualization issues in the old swaps

File: src/store/getters.js (hunk 2 above)
Comment (tima-t) [reply src/store/getters.js:96]: Yes, they are doing the same and V3 will be deprecated soon . 

I need this line, because we have swaps in the history that have been using V3 and they are in the local storage. 
If we do not have a backward support this will result in visualization issues in the old swaps. 

I can also leave the old oneinchV3 and supported v3 and v4, but it does not make much sense to me

**Review by skeremidchiev — APPROVED** [review]
